### DISCUSSION TITLE
TeleBot.send_contact shares contact which is greyed out

### DISCUSSION THREAD

**Discussion by Slava1793** [eternnoir/pyTelegramBotAPI#1954] [Q&A]
1. pyTelegramBotAPI==4.8.0
2. MacOS
3. Python 3.10

When trying to share contact with TeleBot.send_contact (even if user has shared his telephone number via KeyboardButton(request_contact=True)) sometimes the shared contact is greyed out and user has no option to write him directly but only has options to invite to telegram or to add to contacts. I believe that it might be connected with privacy of telegram user and lifetime of bot session and request_contact, but unfortunately I haven't found anything in telegram API docs. Could you please clarify this point?

Thanks

**Comment by Badiboy** [comment]
Such behaviour is on the Telegram side and is out of library business...

**Reply by Slava1793** [reply]
Hi @Badiboy, thanks for your answer.

The question in domain of library: whether or not user's contact lifetime shared in bot session is scoped by lifetime of session? I mean could it expire if bot session expires? And connected question is there any documentation regarding session lifecycle?

**Comment by Badiboy** [comment]
No any "sessions" neiter in TG, bot or library AFAIK.
